ci-based/zeek_benchmarker/tasks.py

```diff
@@ -115,30 +121,29 @@ def unpack_build(
             ]
         )
 
-        args = [
-            "/usr/bin/docker",
-            "run",
-            "--rm",
-            "--security-opt",
-            "no-new-privileges",
-            "--network",
-            "none",
-            "-t",
-            "--mount",
-            shlex.quote(f"type=volume,source={volume},destination={target_dir}"),
-            "--mount",
-            shlex.quote(f"type=bind,source={build_file_path},destination={source_dir}"),
-            "-w",
-            shlex.quote(source_dir),
-            image,
-            "bash",
-            "-x",
-            "-c",
-            tar_command,
+        mounts = [
+            docker.types.Mount(
+                read_only=True,
+                target=source_dir,
+                source=build_file_path,
+                type="bind",
+            ),
+            docker.types.Mount(
+                target=target_dir,
+                source=volume,
+            ),
         ]
 
-        logger.debug("Unpack: %s", " ".join(args))
-        subprocess.run(args, check=True, timeout=timeout + 5)
+        logger.debug("Unpacking %s in container: %s", build_path, command)
+        self._client.containers.run(
+            image=image,
+            mounts=mounts,
+            auto_remove=True,
+            network_disabled=True,
+            working_dir=source_dir,
+            security_opt=["no-new-privileges"],
+            command=["bash", "-x", "-c", command],
+        )
```

Comment:
I can see this being better than the docker-compose, i guess

Reply:
I've re-worked running the benchmarker container using the Docker API directly. A bit of a draw-back is that the image which runs the benchmarks now has to be built, but not the end of the world IMO.